### ISSUE TITLE
Ds interfaces are documented as classes

### ISSUE THREAD

**Issue by AndrolGenhald** [php/doc-en#1534]
https://www.php.net/manual/en/class.ds-collection.php
https://www.php.net/manual/en/class.ds-hashable.php
https://www.php.net/manual/en/class.ds-sequence.php

Both the URLs and the Interface synopses incorrectly indicate that these are classes (although it appears [other](https://www.php.net/manual/en/class.jsonserializable.php) interface URLs also say class, so maybe part is intentional?).

**Comment by cmb69** [comment]
The `class.` prefix in the URL is intentional (classes and interface live in the same namespace anyway). 

But, yeah, the markup for the interfaces is broken (and also interfaces are supposed to extend other interfaces, not to implement them).

**Comment by mallardduck** [comment]
@cmb69 - I don't know enough about phd and the few hours I've been playing with it have been fruitless.

However, from the XML side of things I quickly found that this is valid and gets closer to what we'd desire here:

```xml
    <classsynopsisinfo>

     <oointerface>
      <interfacename>Ds\Collection</interfacename>
     </oointerface>

     <oointerface>
      <modifier>extends</modifier>
      <interfacename>Traversable</interfacename>
     </oointerface>

     <oointerface>
      <interfacename>Countable</interfacename>
     </oointerface>

     <oointerface>
      <interfacename>JsonSerializable</interfacename>
     </oointerface>

    </classsynopsisinfo>
```

The issue is that `phd` still renders the initial `implements` rather than only rendering `extends`. So this is the end result:

<img width="783" alt="Screen Shot 2022-04-21 at 10 12 47 PM" src="https://user-images.githubusercontent.com/619938/164582844-1b0588a7-f054-4707-80a7-f2db7affb0d5.png">

**Comment by cmb69** [comment]
@mallardduck, yeah, this isn't properly solveable with current PhD. [gen_stub.php](https://github.com/php/php-src/blob/master/build/gen_stub.php) would generate the following:
````xml
    <classsynopsisinfo>

     <oointerface>
      <interfacename>Ds\Collection</interfacename>
     </oointerface>

     <ooclass>
      <modifier>extends</modifier>
      <classname>Traversable</classname>
     </ooclass>
     <ooclass>
      <classname>Countable</classname>
     </ooclass>
     <ooclass>
      <classname>JsonSerializable</classname>
     </ooclass>

    </classsynopsisinfo>
````
This results in:

![Screenshot 2022-04-22 113346](https://user-images.githubusercontent.com/2306138/164679342-989d496e-4c3e-49b3-83b1-d633932f5394.gif)

So, besides the markup is not quite correct, the commas between the interface names are missing.

We need to fix PhD in this regard. We also should check [docgen](https://github.com/php/doc-base/tree/master/scripts/docgen), and it may make sense to add a [skeleton](https://github.com/php/doc-base/tree/master/RFC/skeletons) for interfaces.

**Comment by mallardduck** [comment]
Wasn't exactly sure the best place to start, so I just decided to pick the parts that are most familiar to me currently. I figure that's the best way to make sense out of these new things I'm coming across.

As such I've started by making a PR for the interface skeleton: https://github.com/php/doc-base/pull/61

**Comment by Girgias** [comment]
Fixed via #2549
